File: isma-lisma/src/main/java/ru/nstu/isma/in/lisma/analysis/parser/InternalConstUtil.java
package ru.nstu.isma.in.lisma.analysis.parser;

import ru.nstu.isma.in.lisma.analysis.gen.LismaParser;

/**
 * Created by Bessonov Alex
 * on 29.04.2015.
 */
public class InternalConstUtil {

    public static Double getConst(String name) {
        switch (name.toUpperCase()) {
            case "PI":
                return Math.PI;
            case "E":
                return Math.E;
            case "G":
                return 9.81;
        }
        return null;
    }

    public static Double getSpatialBound(LismaParser.Spatial_var_boundContext ctx) {
        Double o = getConst(ctx.getText());
        if (o != null)
            return o;
        return Double.valueOf(ctx.getText());
    }

}

File: isma-lisma/src/main/kotlin/ru/nstu/isma/lisma/parser/InternalConstUtil.kt
package ru.nstu.isma.lisma.parser

import ru.nstu.isma.lisma.analysis.gen.LismaParser.Spatial_var_boundContext
import ru.nstu.isma.lisma.parser.InternalConstUtil
import java.util.*

/**
 * Created by Bessonov Alex
 * on 29.04.2015.
 */
object InternalConstUtil {
    @JvmStatic
    fun getConst(name: String): Double? {
        when (name.uppercase(Locale.getDefault())) {
            "PI" -> return Math.PI
            "E" -> return Math.E
            "G" -> return 9.80665
        }
        return null
    }

    @JvmStatic
    fun getSpatialBound(ctx: Spatial_var_boundContext): Double {
        val o = getConst(ctx.text)
        return o ?: java.lang.Double.valueOf(ctx.text)
    }
}